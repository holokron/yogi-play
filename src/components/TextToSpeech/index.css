.input-rounded-left {
    border-radius: 30px 0 0 30px !important;
}

.button-rounded-right {
    border-radius: 0 30px 30px 0 !important;
}