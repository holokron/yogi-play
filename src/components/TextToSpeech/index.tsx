import * as React from 'react'
import { connect } from 'react-redux'
import { Form, Input, InputGroup, InputGroupAddon, Button, Nav } from 'reactstrap'
import './index.css'
import { readText, AppDispatch } from '../../store/actions'

export interface Props {
    readText: {(text: string): void}
}

export interface State {
    text: string | null
}

class TextToSpeech extends React.PureComponent<Props, State> {
    public state: State = {
        text: null,
    }

    constructor(props: Props) {
        super(props)

        this.handleChange = this.handleChange.bind(this)
        this.handleClick = this.handleClick.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    public handleChange(event: React.ChangeEvent<HTMLInputElement>): void {
        event.preventDefault()

        this.setState({
            text: event.currentTarget.value,
        })
    }

    public handleClick(event: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>): void {
        event.preventDefault()

        this.readText()
    }

    public handleSubmit(event: React.FormEvent<HTMLFormElement>): void {
        event.preventDefault()

        this.readText()
    }

    public render() {
        return (
            <Nav navbar className="mr-auto">
                <Form onSubmit={this.handleSubmit} inline noValidate>
                    <InputGroup>
                        <Input
                            bsSize="sm"
                            className="input-rounded-left"
                            onChange={this.handleChange}
                            placeholder="wpisz tekst"
                        />
                        <InputGroupAddon addonType="append">
                            <Button
                                className="button-rounded-right"
                                color="primary"
                                onClick={this.handleClick}
                                size="sm"
                            >
                                Czytaj
                            </Button>
                        </InputGroupAddon>
                    </InputGroup>
                </Form>
            </Nav>
        )
    }

    private readText(): void {
        if (!this.state.text) {
            return
        }

        this.props.readText(this.state.text)
    }
}

const mapDispatchToProps = (dispatch: AppDispatch) => ({
    readText: (text: string): void => {
        dispatch(readText(text))
    },
})

export default connect(null, mapDispatchToProps)(TextToSpeech)