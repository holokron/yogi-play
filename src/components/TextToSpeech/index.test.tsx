import * as React from "react";
import * as renderer from "react-test-renderer";
import TextToSpeech, { TextToSpeechProps } from ".";

describe("@components/TextToSpeech", () => {
  it.each([
    [
      {
        readText: (): void => {},
      },
    ],
  ] as TextToSpeechProps[][])(
    "renders correctly with props: %p",
    (props: TextToSpeechProps) => {
      const tree = renderer.create(<TextToSpeech {...props} />).toJSON();

      expect(tree).toMatchSnapshot();
    }
  );
});
