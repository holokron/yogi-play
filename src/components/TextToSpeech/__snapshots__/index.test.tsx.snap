// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/TextToSpeech renders correctly with props: {"readText": [Function readText]} 1`] = `
<ul
  className="navbar-nav"
>
  <form
    className="form-inline"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <input
        className="input-rounded-left form-control-sm form-control"
        onChange={[Function]}
        placeholder="wpisz tekst"
        type="text"
      />
      <div
        className="input-group-append"
      >
        <button
          aria-label={null}
          className="button-rounded-right btn btn-primary btn-sm"
          onClick={[Function]}
          type="button"
        >
          Czytaj
        </button>
      </div>
    </div>
  </form>
</ul>
`;
